functions/src/models/requests/timeline.ts

```diff
@@ -0,0 +1,96 @@
+import { FirestoreDataConverter } from '@google-cloud/firestore';
+import { IsEnum, IsObject } from 'class-validator';
+import { firestore } from 'firebase-admin';
+
+import { IOffer } from '../offers';
+import { IRequest, RequestStatus } from './index';
+import Timestamp = firestore.Timestamp;
+import DocumentData = firestore.DocumentData;
+
+export interface ITimelineItem extends DocumentData {
+  offerSnapshot: IOffer;
+  requestSnapshot: IRequest;
```

Comment:
I'm assuming that from the file path, you're planning on having these routed as sub-collections under requests, which allows us to easily query the timeline for a single request, but not across requests.

This makes sense to me .

Reply:
Yes, it's a sub of each request so you can ask a request for it's timeline and all it's offers :)